frontend/components/UpdateAgentPage/context/UpdateAgentProvider.tsx

```diff
@@ -43,32 +44,42 @@ export const UpdateAgentProvider = ({ children }: PropsWithChildren) => {
 
     if (!selectedService || !selectedService.service_config_id) return;
 
-    try {
-      await ServicesService.updateService({
-        serviceConfigId: selectedService.service_config_id,
-        partialServiceTemplate: {
-          ...formValues,
-          env_variables: {
-            ...Object.entries(formValues.env_variables ?? {}).reduce(
-              (acc, [key, value]) => ({
-                ...acc,
-                [key]: {
-                  // Pass the environment variable details
-                  // in case the variable doesn't exist yet in the service
-                  ...(SERVICE_TEMPLATES.find(
-                    (template) =>
-                      template.name === selectedService.name ||
-                      template.agentType === selectedAgentType,
-                  )?.env_variables?.[key] || {}),
-                  // Update with the value from the form
-                  value,
-                },
-              }),
-              {},
-            ),
-          },
+    const currentTemplate = SERVICE_TEMPLATES.find(
+      ({ name, agentType }) =>
+        name === selectedService.name || agentType === selectedAgentType,
+    );
+
+    // TODO: This should be in MemesUpdatePage and not here
+    // Better approach would be to pass formValues as a argument to the function
+    if (selectedAgentType === AgentType.Memeooorr) {
```

Comment:
maybe we can simply compare the form fields names with the service template's env variables and remove what's not there? wdyt? as an agent agnostic approach